sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -499,9 +500,8 @@ public Read withConnectionConfiguration(ConnectionConfiguration connectionConfig
      *     DSL</a>
      * @return a {@link PTransform} reading data from Elasticsearch.
      */
-    public Read withQuery(String query) {
+    public Read withQuery(ValueProvider<String> query) {
```

Comment:
Hello!  For compatibility, what do you think about keeping the original `withQuery` signature (i.e. an overloaded method that takes a String _or_ a ValueProvider<String>)?

This is a really common pattern: https://github.com/apache/beam/blob/c2f0d282337f3ae0196a7717712396a5a41fdde1/sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java#L744

Reply:
Hi @RyanSkraba. 
Thanks for taking a look. Yes I should have done as you suggested in the first place to avoid peoples code breaking because of my change. Pushed my changes.